Remotion/Web/UnitTests/Core/ExecutionEngine/Infrastructure/ScopedTransactionStrategyTests/Common.cs

```diff
@@ -16,151 +16,163 @@
   {
-    private ScopedTransactionStrategyBase _strategy;
+    private Mock<ScopedTransactionStrategyBase> _strategy;
 
     public override void SetUp ()
     {
       base.SetUp();
-      ExecutionContextMock.BackToRecord();
-      ExecutionContextMock.Stub (stub => stub.GetInParameters()).Return (new object[0]).Repeat.Any();
-      ExecutionContextMock.Replay();
+      ExecutionContextMock.Setup (stub => stub.GetInParameters()).Returns (new object[0]);
 
-      TransactionMock.BackToRecord();
-      TransactionMock.Stub (stub => stub.EnsureCompatibility (Arg<IEnumerable>.Is.NotNull));
-      TransactionMock.Replay();
+      TransactionMock.Setup (stub => stub.EnsureCompatibility (It.IsNotNull<IEnumerable>()));
 
-      _strategy = MockRepository.PartialMock<ScopedTransactionStrategyBase> (
-          true, (Func<ITransaction>) (() => TransactionMock), OuterTransactionStrategyMock, ExecutionContextMock);
-      _strategy.Replay();
-
-      ExecutionContextMock.BackToRecord();
-      TransactionMock.BackToRecord();
+      _strategy = new Mock<ScopedTransactionStrategyBase> (
+          true, (Func<ITransaction>) (() => TransactionMock.Object), OuterTransactionStrategyMock.Object, ExecutionContextMock.Object) { CallBase = true };
     }
 
     [Test]
     public void Initialize ()
     {
-      Assert.That (_strategy.ExecutionContext, Is.SameAs (ExecutionContextMock));
-      Assert.That (_strategy.AutoCommit, Is.True);
-      Assert.That (_strategy.IsNull, Is.False);
-      Assert.That (_strategy.OuterTransactionStrategy, Is.SameAs (OuterTransactionStrategyMock));
-      Assert.That (_strategy.Child, Is.SameAs (NullTransactionStrategy.Null));
+      Assert.That (_strategy.Object.ExecutionContext, Is.SameAs (ExecutionContextMock.Object));
+      Assert.That (_strategy.Object.AutoCommit, Is.True);
+      Assert.That (_strategy.Object.IsNull, Is.False);
+      Assert.That (_strategy.Object.OuterTransactionStrategy, Is.SameAs (OuterTransactionStrategyMock.Object));
+      Assert.That (_strategy.Object.Child, Is.SameAs (NullTransactionStrategy.Null));
     }
 
     [Test]
     public void GetTransaction ()
     {
-      TransactionMock.Expect (mock => mock.To<ITransaction>()).Return (TransactionMock);
-      TransactionMock.Replay();
-      Assert.That (_strategy.GetNativeTransaction<ITransaction>(), Is.SameAs (TransactionMock));
+      TransactionMock.Setup (mock => mock.To<ITransaction>()).Returns (TransactionMock.Object).Verifiable();
+      Assert.That (_strategy.Object.GetNativeTransaction<ITransaction>(), Is.SameAs (TransactionMock.Object));
     }
 
     [Test]
     public void Commit ()
     {
-      TransactionMock.Expect (mock => mock.Commit());
-      MockRepository.ReplayAll();
+      TransactionMock.Setup (mock => mock.Commit()).Verifiable();
 
-      _strategy.Commit();
+      _strategy.Object.Commit();
 
-      MockRepository.VerifyAll();
+      TransactionFactoryMock.Verify();
```

Comment:
It lookls like the base-type already has a VerifyAll Method we could use instead of adding individual verifies for the mocks declared by the base type. please refactor.